test/e2e/specs/all-features-confirmation/errors/errors.spec.js

```diff
@@ -337,107 +289,107 @@ describe('Error related test cases,', function() {
               return modalTitle.getText();
           }).then(function (text) {
               expect(text).toBe("Confirm Delete", "Deleteion confirmation pop-up could not be opened!");
-              chaisePage.recordPage.getConfirmDeleteButton().click();
-              errModalClass =  chaisePage.recordPage.getModalText();
-              chaisePage.waitForElement(errModalClass);
-              return errModalClass.getText();
+              return chaisePage.recordPage.getConfirmDeleteButton().click().then(function() {
```

Comment:
It looks like you are intending to write this as a promise chain by returning the promise each time, but you are acting on the promise right away rather than part of the outer chain. This should be changed to something like:
```js
  return chaisePage.recordPage.getConfirmDeleteButton().click();
}).then(function() {
  errModalClass =  chaisePage.recordPage.getModalText();
  return chaisePage.waitForElement(errModalClass);
}).then(function() {
  return errModalClass.getText();
```